### PR TITLE
Sending validation success and failure emails from orchestrator

### PR DESCRIPTION
**Pull request by agr**
Addresses https://github.com/NuGet/NuGetGallery/issues/4659

### CODE REVIEW

File: src/NuGet.Services.Validation.Orchestrator/EmailConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Services.Validation.Orchestrator
+{
+    public class EmailConfiguration
+    {
+        public string GalleryOwner { get; set; }
+        public string GalleryNoReplyAddress { get; set; }
+        public string PackageUrlTemplate { get; set; }
```
Comment (loic-sharma) [inline src/NuGet.Services.Validation.Orchestrator/EmailConfiguration.cs]: Please add comments about the template parameters. IE: {0} is ..., {1} is ....

File: src/NuGet.Services.Validation.Orchestrator/NullMailSender.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+using System.Net.Mail;
+using AnglicanGeek.MarkdownMailer;
+
+namespace NuGet.Services.Validation.Orchestrator
+{
+    /// <summary>
+    /// <see cref="IMailSender"/> implementation that does nothing for the case when email is not configured
+    /// </summary>
+    public class NullMailSender : IMailSender
```
Comment (joelverhagen) [inline src/NuGet.Services.Validation.Orchestrator/NullMailSender.cs]: Why allow this case? Our services will always have this configured, right? Or does DEV use this?

Comment (scottbommarito) [reply src/NuGet.Services.Validation.Orchestrator/NullMailSender.cs]: For use while debugging probably. DEV should have it configured properly.

File: src/NuGet.Services.Validation.Orchestrator/EmailConfiguration.cs (hunk 1 above)
Comment (scottbommarito) [reply src/NuGet.Services.Validation.Orchestrator/EmailConfiguration.cs]: Comments on how each configuration value is used would be ideal.

File: src/NuGet.Services.Validation.Orchestrator/NullMailSender.cs (hunk 2 above)
Comment (agr) [reply src/NuGet.Services.Validation.Orchestrator/NullMailSender.cs]: Yeah, I want to run it on my dev box without sending any stuff out and without failing on invalid configuration, invalid credentials or invalid host names.

Comment (joelverhagen) [reply src/NuGet.Services.Validation.Orchestrator/NullMailSender.cs]: In gallery we have a file system writer, which allows you to see the actual email.

File: src/NuGet.Services.Validation.Orchestrator/EmailConfiguration.cs (hunk 1 above)
Comment (agr) [reply src/NuGet.Services.Validation.Orchestrator/EmailConfiguration.cs]: Done

File: src/NuGet.Services.Validation.Orchestrator/NullMailSender.cs (hunk 2 above)
Comment (agr) [reply src/NuGet.Services.Validation.Orchestrator/NullMailSender.cs]: Done

**Comment by agr** [comment]
Validated on dev

**Pull request merged**